samples/algorithms/GHZ.qs

```diff
@@ -30,23 +31,18 @@ namespace Sample {
         DumpMachine();
 
         // Measure and reset qubits before releasing them.
-        let results = MeasureEachZ(qs);
-        ResetAll(qs);
-        return results;
+        MResetEachZ(qs)
     }
 
     /// # Summary
-    /// This operation prepares a generalized GHZ state across a register of qubits.
-    ///
-    /// # Input
-    /// ## qs
-    /// The given register of qubits to be transformed into the GHZ state. It is assumed
-    /// that these qubits are in their default |0〉 state.
+    /// Prepares state (|000〉 + |111〉) / √2 (GHZ state) across a register
+    /// of three qubits `qs`.
+    /// All qubits are assumed to be in |0〉 state on input.
     operation PrepareGHZState(qs : Qubit[]) : Unit {
-        Fact(Length(qs) > 0, "`qs` length must be greater than zero");
-        Fact(CheckAllZero(qs), "All qubits must be in the |0〉 state");
+        Fact(Length(qs) == 3, "`qs` length be 3.");
 
         H(qs[0]);
-        ApplyToEach(CNOT(qs[0], _), qs[1...]);
+        CNOT(qs[0], qs[1]);
+        CNOT(qs[0], qs[2]);
     }
```

Comment:
Rather than make this specific to the length 3, this could remain generalized and avoid `ApplyToEach` like this:
```qsharp
    operation PrepareGHZState(qs : Qubit[]) : Unit {
        Fact(Length(qs) >= 2, "`qs` length should be at least 2.");

        H(qs[0]);
        for q in qs[1...] {
            CNOT(qs[0], q);
        }
    }
```

Reply:
I think it's ok to leave it in 3 since GHZ is specifically a three-qubit state. The more general case is covered by the cat state sample.

Reply:
Yes, just like Cesar mentioned I prefer to have specific implementation for 3 qubits here in GHZ sample because Cat sample has general implementation.